app/models/order_item.rb

```diff
@@ -92,4 +94,9 @@ def mark_finished
     self.state = 'Completed'
     save!
   end
+
+  def workflow_ref
+    hierarchy_search = Catalog::HierarchicalModelSearch.new("workflow_ref", portfolio_item.id, %w[PortfolioItem Portfolio])
```

Comment:
I imagined this would be something like a `resolved_workflow_ref` method on the target object, in this case `portfolio_item`.  That way we do not override the `workflow_ref` property on the `portfolio_item` and the logic can be customized to that object if needed.  It keeps the logic with the object it is associated with.

Then the order_item just asks the object what workflow to run and does not have to know the details of how to look it up.

Thoughts?

Reply:
Yep it's looking like I misunderstood a bit - I thought it was supposed to be returned in a response, but this will actually make the PR smaller a bit since I won't have to touch `OrderItem` really.